### PR TITLE
[SYCL] Emit program build logs for warning levels >= 2

### PR DESCRIPTION
**Pull request by alexbatashev**
When `SYCL_RT_WARNING_LEVEL` is set to 2 or higher, a program build log will be printed to `std::clog`.

### CODE REVIEW

File: sycl/source/detail/kernel_program_cache.hpp [hunk 1]
```diff
@@ -130,6 +130,12 @@ class KernelProgramCache {
     MKernelFastCache.emplace(CacheKey, CacheVal);
   }
 
+  void reset() {
```
Comment (s-kanaev) [inline sycl/source/detail/kernel_program_cache.hpp:133]: A comment saying that this method is for testing purposes only might be useful here.

**Comment by alexbatashev** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/753

**Comment by alexbatashev** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/753

**Comment by alexbatashev** [comment]
@s-kanaev ping

File: sycl/source/detail/program_manager/program_manager.cpp [hunk 2]
```diff
@@ -438,6 +438,17 @@ std::pair<RT::PiProgram, bool> ProgramManager::getOrCreatePIProgram(
   return {NativePrg, BinProg.size()};
 }
 
+/// Emits information about built programs if the appropriate contitions are
+/// met, namely when SYCL_RT_WARNING_LEVEL is greater than or equal to 2.
+static void emitBuiltProgramInfo(const pi_program &Prog,
+                                 const ContextImplPtr &Context) {
+  if (SYCLConfig<SYCL_RT_WARNING_LEVEL>::get() >= 2) {
```
Comment (s-kanaev) [inline sycl/source/detail/program_manager/program_manager.cpp:445]: NIT: `2` got to have a name here. Suggest add naming in separate patch when we're to add some behaviour for another value e.g. `1`.

**Review by s-kanaev — APPROVED** [review]
LGTM

**Comment by bader** [comment]
Verification with https://github.com/intel/llvm-test-suite/pull/753 fails on Windows. @alexbatashev, please, fix regressions.

**Comment by alexbatashev** [comment]
@bader I don't see how this patch could have caused that. The same tests pass on OpenCL backend for GPUs and CPUs, and only fail on L0. I can disable these tests and open a tracker.

**Comment by bader** [comment]
SYCL :: Basic/image/image_accessor_readsampler.cpp is failing. Is it related to this patch?

**Pull request merged**